zenoh/tests/qos.rs

```diff
@@ -0,0 +1,45 @@
+use std::time::Duration;
+
+use async_std::task;
+use zenoh::{publication::Priority, SessionDeclarations};
+use zenoh_core::{zasync_executor_init, AsyncResolve};
+use zenoh_protocol::core::CongestionControl;
+
+const SLEEP: Duration = Duration::from_secs(1);
+
+#[test]
+fn pubsub() {
+    task::block_on(async {
+        zasync_executor_init!();
+        let session1 = zenoh::open(zenoh_config::peer()).res_async().await.unwrap();
+        let session2 = zenoh::open(zenoh_config::peer()).res_async().await.unwrap();
+
+        let publisher = session1
+            .declare_publisher("test/qos")
+            .res()
+            .await
+            .unwrap()
+            .priority(Priority::DataHigh)
+            .congestion_control(CongestionControl::Drop);
+
+        task::sleep(SLEEP).await;
+
+        let sub = session2.declare_subscriber("test/qos").res().await.unwrap();
+        task::sleep(SLEEP).await;
+
+        publisher.put("qos").res_async().await.unwrap();
+        let qos = sub.recv_async().await.unwrap().qos;
```

Comment:
It's better to wait with a timeout with e.g. a [ztimeout!()](https://github.com/eclipse-zenoh/zenoh/blob/84dc04acdcf794e71667e0e76e521f41b921eef8/io/zenoh-transport/tests/unicast_transport.rs#L228) macro.

Reply:
fixed